scvi/_settings.py

```diff
@@ -121,6 +121,10 @@ def seed(self) -> int:
         """Random seed for torch and numpy."""
         return self._seed
 
+    def set_num_threads(self, num: int):
+        """Sets number of threads torch will use to `num`."""
+        torch.set_num_threads(num)
+
```

Comment:
`num_threads` needs to be a property with a setter, the setter runs num threads, you can default the property to return None?